saw-core/src/Verifier/SAW/SATQuery.hs

```diff
@@ -0,0 +1,27 @@
+module Verifier.SAW.SATQuery where
+
+import Control.Monad (foldM)
+import Data.Map (Map)
+import Data.Set (Set)
+
+import Verifier.SAW.Name
+import Verifier.SAW.FiniteValue
+import Verifier.SAW.SharedTerm
+
+data SATQuery =
```

Comment:
Let's put some haddock documentation on this datatype, at least. The `satAsserts` field is a list; that's interpreted as a logical conjunction, I suppose? And it looks like there is an internal invariant on the `satVariables` field, where the `FirstOrderType` values in the map should match the `Term` type on the `ExtCns` keys, I think.